File: src/main/java/Spring.java
public class Spring {
    protected int name;
    protected Mass mass1, mass2;
    protected double restLength;

    public Spring(int name) {
        this.name = name;
    }

    public Spring() {
    }

    public int getName() {
        return name;
    }

    public Mass getMass1() {
        return mass1;
    }

    public void setMass1(Mass m) {
        mass1 = m;
    }

    public Mass getMass2() {
        return mass2;
    }

    public void setMass2(Mass m) {
        mass2 = m;
    }

    public double getRestLength() {
        return restLength;
    }

    public void setRestLength(double r) {
        this.restLength = r;
    }

    public String toString() {
        return name + " a:" + mass1 + " b:" + mass2 + " restlength:" + restLength;
    }

}

File: src/main/kotlin/com/ihd2/model/Spring.kt
package com.ihd2.model

import com.ihd2.graphics.GraphicsRenderer
import com.ihd2.graphics.Renderable
import java.awt.Color

open class Spring(val id: Int): Renderable {
    lateinit var mass1: Mass
    lateinit var mass2: Mass
    var restLength = 0.0
    var hasCollided = false
        set(value) {
            field = value
            mass1.hasCollided = value
            mass2.hasCollided = value
        }

    /**
     * Calculates the rest length at the current frame
     */
    open fun getRestLength(model: Model) = restLength

    override fun render(color: Color, renderer: GraphicsRenderer) {
        val colorToUse = when (renderer.isDebug()) {
            true -> if (hasCollided) Color.RED else color
            false -> color
        }
        renderer.drawLine(
            colorToUse,
            mass1.position.x,
            mass1.position.y,
            mass2.position.x,
            mass2.position.y)
        hasCollided = false
    }
}